modules/fb_instant_articles_display/fb_instant_articles_display.module

```diff
@@ -0,0 +1,777 @@
+    }
+    foreach ($entity_types as $entity_type => $bundles) {
+      foreach ($bundles as $bundle_type => $bundle) {
+        $form_name = $bundle_type . '_node_form';
+        if ($form_id == $form_name) {
+          // Add the module's form settings to this individual node's form.
+          // Add a vertical tab to the node type form.
+          if (user_access('administer fb_instant_articles_display')) {
+            // Build fieldset for vertical tab section.
+            $fieldset = array(
+              '#type' => 'fieldset',
+              '#title' => t('Facebook Instant Articles'),
+              '#description' => t('Configure entity for Facebook Instant Article mappings.'),
+              '#group' => 'additional_settings',
+              '#collapsible' => TRUE,
+              '#collapsed' => TRUE,
+            );
+
+            // Has the section already been created (perhaps by a sub module)?
+            if (isset($form['fb_instant_articles_display'])) {
+              // Add additional fieldset data.
+              $form['fb_instant_articles_display'] += $fieldset;
+            }
+            // Else, create the fieldset.
+            else {
+              $form['fb_instant_articles_display'] = $fieldset;
+            }
+
+            // Is an Article?
+            $type = $form['#node']->type;
+            $fb_instant_enabled = fb_instant_articles_display_is_article($form['#node']->nid);
+
+            // Previously checked the product checkbox?
+            $previously_checked = (isset($form_state['values']) && $form_state['values']['fb_instant_articles_display']['fb_instant_enabled']);
+
+            // Build the product checkbox.
+            $form['fb_instant_articles_display']['fb_instant_enabled'] = array(
+              '#type' => 'checkbox',
+              '#title' => t('Include entity in Facebook Instant Articles feed.'),
+              '#description' => t('Enable this entity to be included in the Facebook Instant Articles feed.'),
+              '#weight' => 0,
+              '#default_value' => ($previously_checked || $fb_instant_enabled) ? TRUE : FALSE,
+            );
+
+            // Add custom submit.
+            $form['#submit'][] = 'fb_instant_articles_display_node_form_submit';
+          }
+          // Since only one node add/edit form can (probably?) be loaded on a
+          // page at a time, we can break out of this loop.
+          break;
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Submit callback for node form.
+ */
+function fb_instant_articles_display_node_form_submit($form, &$form_state) {
+  $fb_instant_enabled = fb_instant_articles_display_is_article_type('node', $form['type']['#value']);
+  if ($fb_instant_enabled) {
+    if (!isset($form_state['values']['fb_instant_enabled'])) {
+      $form_state['values']['fb_instant_enabled'] = 0;
+    }
+    $value = $form_state['values']['fb_instant_enabled'];
+    $default_enabled = variable_get('fb_instant_articles_display_all_enabled_default', TRUE);
```

Comment:
Hm yes I missed that one - nice catch. So… this throws a small wrench in my desire to change as little of this existing code as possible in this PR. Personally, I think all these settings really do have more to do with RSS submodule functionality than the Display-related functionality (because they determine which entities will show in the feed page, whereas the Views submodule can let site builders choose any method they wish for selecting which items get added to the Views-controlled feed: a custom checkbox field, date ranges, or other complex selection logic). This is why I added these settings to the RSS module's settings form – but yes I agree it doesn't make sense to have this simple selection functionality in the Display module but it's configs in the RSS module (since the dependency is the other way round). By this logic, perhaps these functions that relate to simple selection of feed items should all move to the RSS module. Does this make sense?